### PR TITLE
Update disaster recovery guidance

### PR DESCRIPTION
**Pull request by RMcVelia**
Update disaster recovery guidance

https://trello.com/c/g4YUcnbV/1860-dr-update-tech-guidance

### CODE REVIEW

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 1]
```diff
@@ -49,13 +49,23 @@ It is possible to lose the database instance and the associated backups. For exa
 |Detection|Endpoint monitoring may point to a healthcheck page checking the connection to the databse. Or smoke tests running in production may detect it.|
 |Remediation|Restore database from external daily or most recent backup|
 
+## Accidental resource deletion
```
Comment (gunndabad) [inline source/infrastructure/disaster-recovery/index.html.md.erb:52]: I wonder whether we could use something like this on DB resources too: https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#prevent_destroy

**Review by gunndabad — APPROVED** [review]

**Comment by RMcVelia** [comment]
[DR - update tech guidance](https://trello.com/c/g4YUcnbV/1860-dr-update-tech-guidance)

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 2]
```diff
@@ -7,7 +7,7 @@ weight: 40
 
 <%= partial('partials/page_toc') %>
 
-This document is intended to list technical risks to our digital services and the mitigations we have in place.
+This document is intended to list technical risks to our digital services and the mitigations we have in place.<br/><br/>For any issue affecting a service (or services) always check if there are any dependant services that may also be affected [Teacher Services Dependencies](https://educationgovuk.sharepoint.com/sites/teacher-services-infrastructure/SitePages/Teacher-services-dependencies.aspx)
```
Comment (saliceti) [inline source/infrastructure/disaster-recovery/index.html.md.erb:10]: Cosmetic: 
```
always [check if there are any dependant services](https://educationgovuk.sharepoint.com/sites/teacher-services-infrastructure/SitePages/Teacher-services-dependencies.aspx) that may also be affected.
```

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 3]
```diff
@@ -49,13 +49,23 @@ It is possible to lose the database instance and the associated backups. For exa
...
+|Prevention|Applications should be built with failure in mind: deploy multiple application instances. Spread them across multiple AZs for high availability.<br/>Our AKS clusters are spread across 3 AZs. Scale applications to more than 1 replicas and enable zone redundancy.<br/>Azure storage accounts should be ZRS or GZRS if zone redundancy is required (automatic and manual failover).<br/>Azure key vault uses replication within region and to a paired region (automatic failover).<br/>Azure Postgres can be configured zone redundant, with the active/standby instances in different zones.<br/>Azure Redis can be zone redundant only if using the Premium SKU.<br/>Postgres and Redis utilise automatic failover. Postgres can also be failed over manually.<br/>Cluster Public IP addresses should be configured with zone redundancy [Azure PIP redundancy](https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-addresses#availability-zone)|
```
Comment (saliceti) [inline source/infrastructure/disaster-recovery/index.html.md.erb:68]: We could make it a bit more generic: `AKS clusters should be configured with nodes spread across multiple AZS`

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 4]
```diff
@@ -75,6 +85,7 @@ We often rely on Azure for:
 - Terraform state in Azure Storage
 - Infrastructure and application secrets in Azure Key Vault
 - Daily production database backups in Azure Storage
+- some applications use Azure Storage for storing or uploading data
```
Comment (saliceti) [inline source/infrastructure/disaster-recovery/index.html.md.erb:88]: This impacts the running application, not the delivery. It should be covered in the other sections

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 5]
```diff
@@ -171,7 +181,7 @@ If DockerHub is down it won’t impact the running service, but we won’t be ab
 |Remediation|Build and deploy manually|
 
 ## Monitoring and logging failure
-We rely on services like [Logit.io](https://logit.io/), [StatusCake](https://www.statuscake.com/), [Prometheus ecosystem](https://github.com/DFE-Digital/cf-monitoring/), [Skylight](https://www.skylight.io/), [Sentry](https://sentry.io/)
+We rely on services like [Logit.io](https://logit.io/), [StatusCake](https://www.statuscake.com/), [Prometheus ecosystem](https://github.com/DFE-Digital/teacher-services-cloud/), [Skylight](https://www.skylight.io/), [Sentry](https://sentry.io/), [Azure Monitor](https://learn.microsoft.com/en-us/azure/azure-monitor/overview)
```
Comment (saliceti) [inline source/infrastructure/disaster-recovery/index.html.md.erb:184]: Cosmetic: end with full stop.

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 6]
```diff
@@ -191,7 +201,21 @@ GOV.UK Notify is used to communicate with our users via emails, texts and letter
...
 
 ## Google BigQuery
-TBD
+Our services load web analytics and database data into Google BigQuery via an event stream. BQ Data is then available for analysis using various tools.
+
+|||
+|-|-|
+|Impact|Unable to send data to BQ.<br/>Reporting unavailable or out of date.|
+|Prevention||
+|Detection||
```
Comment (saliceti) [inline source/infrastructure/disaster-recovery/index.html.md.erb:210]: Sentry should alert when there is an error while sending to BQ

Comment (saliceti) [reply source/infrastructure/disaster-recovery/index.html.md.erb:210]: Contact Google support (how?)

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 7]
```diff
@@ -191,7 +201,21 @@ GOV.UK Notify is used to communicate with our users via emails, texts and letter
...
 ## Google API
-TBD
+Various Google API's are used by our services, including Geocoding, Indexing, and Analytics.
+
+|||
+|-|-|
+|Impact|Some application functionality will be degraded or unavailable|
+|Prevention||
+|Detection||
+|Remediation||
```
Comment (saliceti) [inline source/infrastructure/disaster-recovery/index.html.md.erb:221]: Contact Google support (how?)

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 2 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: updated

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 3 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: updated

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 4 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: moved

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 5 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: fixed

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 6 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: added more info

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 7 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: added more info

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 8]
```diff
@@ -7,7 +7,7 @@ weight: 40
 
 <%= partial('partials/page_toc') %>
 
-This document is intended to list technical risks to our digital services and the mitigations we have in place.
+This document is intended to list technical risks to our digital services and the mitigations we have in place.<br/><br/>For any issue affecting a service (or services) always [check if there are any dependant services](https://educationgovuk.sharepoint.com/sites/teacher-services-infrastructure/SitePages/Teacher-services-dependencies.aspx) that may also be affected.
```
Comment (carlosmartinez) [inline source/infrastructure/disaster-recovery/index.html.md.erb:10]:
```suggestion
This document is intended to list technical risks to our digital services and the mitigations we have in place.<br/><br/>For any issue affecting a service (or services) always [check if there are any dependent services](https://educationgovuk.sharepoint.com/sites/teacher-services-infrastructure/SitePages/Teacher-services-dependencies.aspx) that may also be affected.
```

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 9]
```diff
@@ -49,13 +49,23 @@ It is possible to lose the database instance and the associated backups. For exa
...
 
+## Accidental resource deletion
+We use terraform to provision resources, but it could be possible for a code change or a user with privileges to delete resources accidentally or otherwise.
+
+|||
+|-|-|
+|Impact|Applications may be unavailable. Data may be lost.|
+|Prevention|Approved PIM request required for production Azure access.<br/>Pull Requests require at least 1 approval.<br/>Soft delete and versioning enabled for key vaults and storage accounts.<br/>Azure resource locks placed on important resources [Azure locks](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/lock-resources?tabs=json). |
+|Detection|Endpoint monitoring may point to a healthcheck page that is now failing. Or smoke tests running in production may detect it.|
+|Remediation|Recovery dependant on the resource deleted, either restore correct version or redeploy and restore data from backup|
```
Comment (carlosmartinez) [inline source/infrastructure/disaster-recovery/index.html.md.erb:60]:
```suggestion
|Remediation|Recovery dependent on the resource deleted, either restore correct version or redeploy and restore data from backup|
```

Comment (carlosmartinez) [reply source/infrastructure/disaster-recovery/index.html.md.erb:60]: sorry! a dependant is a person who is dependent on someone

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 10]
```diff
@@ -49,13 +49,23 @@ It is possible to lose the database instance and the associated backups. For exa
...
+|Prevention|Applications should be built with failure in mind: AKS clusters should be configured with nodes spread across multiple AZS. AKS Deployments should use a zone topology spread constraint. Scale applications to more than 1 replica to enable zone redundancy.<br/>Azure storage accounts should be ZRS or GZRS if zone redundancy is required (automatic and manual failover).<br/>Azure key vault uses replication within region and to a paired region (automatic failover).<br/>Azure Postgres can be configured zone redundant, with the active/standby instances in different zones.<br/>Azure Redis can be zone redundant only if using the Premium SKU.<br/>Postgres and Redis utilise automatic failover. Postgres can also be failed over manually.<br/>Cluster Public IP addresses should be configured with zone redundancy [Azure PIP redundancy](https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-addresses#availability-zone)|
```
Comment (carlosmartinez) [inline source/infrastructure/disaster-recovery/index.html.md.erb:68]:
```suggestion
|Prevention|Applications should be built with failure in mind: AKS clusters should be configured with nodes spread across multiple AZs. AKS Deployments should use a zone topology spread constraint. Scale applications to more than 1 replica to enable zone redundancy.<br/>Azure storage accounts should be ZRS or GZRS if zone redundancy is required (automatic and manual failover).<br/>Azure key vault uses replication within region and to a paired region (automatic failover).<br/>Azure Postgres can be configured zone redundant, with the active/standby instances in different zones.<br/>Azure Redis can be zone redundant only if using the Premium SKU.<br/>Postgres and Redis utilise automatic failover. Postgres can also be failed over manually.<br/>Cluster Public IP addresses should be configured with zone redundancy [Azure PIP redundancy](https://learn.microsoft.com/en-us/azure/virtual-network/ip-services/public-ip-addresses#availability-zone)|
```

File: source/infrastructure/disaster-recovery/index.html.md.erb [hunk 11]
```diff
@@ -65,7 +75,7 @@ In some rare cases, an entire region might become unavailable.
 |||
 |-|-|
 |Impact|Applications may be unavailable|
-|Prevention|For critical applications, it is possible to deploy to 2 different regions, synchronise the data, configure a DNS based failover or GSLB. We don’t usually protect against this risk as it is not worth the complexity of the required set-up.|
+|Prevention|For critical applications, it is possible to deploy to 2 different regions, synchronise the data, configure a DNS based failover or GSLB. We don’t usually protect against this risk as it is not worth the complexity of the required set-up.<br/>Production Postgres backups are kept in a GRS Azure storage account, which maintains copies of data in a separate reqion. Any critcal application data kept in a storage account should be GRS/GZRS.<br/>Azure Key Vault maintains a copy of the contents in another region.<br/>For storage accounts and key vaults, failover is automatic and transparent. Storage accounts also support manual failover. |
```
Comment (carlosmartinez) [inline source/infrastructure/disaster-recovery/index.html.md.erb:78]:
```suggestion
|Prevention|For critical applications, it is possible to deploy to 2 different regions, synchronise the data, configure a DNS based failover or GSLB. We don’t usually protect against this risk as it is not worth the complexity of the required set-up.<br/>Production Postgres backups are kept in a GRS Azure storage account, which maintains copies of data in a separate region. Any critical application data kept in a storage account should be GRS/GZRS.<br/>Azure Key Vault maintains a copy of the contents in another region.<br/>For storage accounts and key vaults, failover is automatic and transparent. Storage accounts also support manual failover. |
```

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 8 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: fixed

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 10 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: fixed

File: source/infrastructure/disaster-recovery/index.html.md.erb (hunk 11 above)
Comment (RMcVelia) [reply source/infrastructure/disaster-recovery/index.html.md.erb]: fixed

**Review by saliceti — APPROVED** [review]

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**